pkg/services/rpcsrv/client_test.go

```diff
@@ -2523,3 +2523,105 @@ func TestGetBlockNotifications(t *testing.T) {
 		require.NotEmpty(t, bn)
 	})
 }
+
+func TestSessionExpansionExtension_SessionDisabled(t *testing.T) {
+	chain, rpcSrv, httpSrv := initClearServerWithCustomConfig(t, func(cfg *config.Config) {
+		cfg.ApplicationConfiguration.RPC.Enabled = true
+		cfg.ApplicationConfiguration.RPC.SessionEnabled = false
+		cfg.ApplicationConfiguration.RPC.SessionExpansionEnabled = true
+		cfg.ApplicationConfiguration.RPC.MaxIteratorResultItems = 5
+	})
+	t.Cleanup(rpcSrv.Shutdown)
+	t.Cleanup(httpSrv.Close)
+
+	for _, b := range getTestBlocks(t) {
+		require.NoError(t, chain.AddBlock(b))
+	}
+
+	c, err := rpcclient.New(context.Background(), httpSrv.URL, rpcclient.Options{})
+	require.NoError(t, err)
+	require.NoError(t, c.Init())
+	t.Cleanup(c.Close)
+
+	storageHash, err := util.Uint160DecodeStringLE(storageContractHash)
+	require.NoError(t, err)
+
+	invokeRes, err := c.InvokeFunction(storageHash, "iterateOverValues", []smartcontract.Parameter{}, nil)
+	require.NoError(t, err)
+	require.Equal(t, vmstate.Halt.String(), invokeRes.State)
+	require.NotEmpty(t, invokeRes.Script)
+
+	require.Len(t, invokeRes.Stack, 1)
+	stackItem := invokeRes.Stack[0]
+	require.Equal(t, stackitem.InteropT, stackItem.Type())
+
+	iterVal, ok := stackItem.Value().(result.Iterator)
+	require.True(t, ok)
+
+	require.Equal(t, len(iterVal.Values), 5)
+	require.True(t, iterVal.Truncated)
```

Comment:
Check that sessionID is not set in the resulted iterator. Check iterator ID as far.